apps/etterlatte-saksbehandling-ui/client/src/components/behandling/beregningsgrunnlag/Beregningsgrunnlag.tsx

```diff
@@ -1,55 +1,65 @@
 import { formaterStringDato } from '~utils/formattering'
 import { hentBehandlesFraStatus } from '../felles/utils'
 import { NesteOgTilbake } from '../handlinger/NesteOgTilbake'
 import { useAppSelector } from '~store/Store'
 import { opprettEllerEndreBeregning } from '~shared/api/beregning'
+import { isFailure, isPendingOrInitial, isSuccess, useApiCall } from '~shared/hooks/useApiCall'
+import { hentSoeskenMedIBeregning, lagreSoeskenMedIBeregning } from '~shared/api/grunnlag'
+import { SoeskenMedIBeregning } from '~shared/types/Grunnlagsopplysning'
+import Spinner from '~shared/Spinner'
+import { IPdlPerson } from '~shared/types/Person'
 
-interface SoeskenMedIBeregning {
+interface FormValues {
   foedselsnummer: string
-  skalBrukes: boolean
+  skalBrukes?: boolean
 }
 
 const Beregningsgrunnlag = () => {
   const { next } = useBehandlingRoutes()
   const behandling = useAppSelector((state) => state.behandlingReducer.behandling)
   const behandles = hentBehandlesFraStatus(behandling?.status)
   const [isLoading, setIsLoading] = useState(false)
+  const [beregningsgrunnlag, hentBeregningsgrunnlag] = useApiCall(hentSoeskenMedIBeregning)
+  const { control, handleSubmit, setValue } = useForm<{ beregningsgrunnlag: FormValues[] }>()
+
+  useEffect(() => {
+    hentBeregningsgrunnlag(behandling.sak, (result) => {
+      setValue('beregningsgrunnlag', result.opplysning.beregningsgrunnlag)
+    })
+  }, [])
 
   if (behandling.kommerBarnetTilgode == null || behandling.familieforhold?.avdoede == null) {
     return <div style={{ color: 'red' }}>Familieforhold kan ikke hentes ut</div>
   }
 
-  const soeker = behandling.søker
-  const soesken = behandling.familieforhold.avdoede.opplysning.avdoedesBarn?.filter(
-    (barn) => barn.foedselsnummer !== soeker?.foedselsnummer
-  )
-  const beregningsperiode = behandling.beregning?.beregningsperioder ?? []
-
-  const { control, handleSubmit } = useForm<{ beregningsgrunnlag: SoeskenMedIBeregning[] }>({
-    defaultValues: {
-      beregningsgrunnlag:
-        soesken?.map((person) => ({
-          foedselsnummer: person.foedselsnummer,
-          skalBrukes: !!beregningsperiode[0]?.soeskenFlokk?.find((fnr) => fnr === person.foedselsnummer),
-        })) ?? [],
-    },
-  })
+  const soesken: IPdlPerson[] =
+    behandling.familieforhold.avdoede.opplysning.avdoedesBarn?.filter(
+      (barn) => barn.foedselsnummer !== behandling.søker?.foedselsnummer
+    ) ?? []
 
-  const lagBeregning = () => opprettEllerEndreBeregning(behandling.id).then(() => next())
+  const onSubmit = async (beregningsgrunnlag: SoeskenMedIBeregning[]) => {
+    setIsLoading(true)
+    await lagreSoeskenMedIBeregning(behandling.id, beregningsgrunnlag)
+      .then(() => opprettEllerEndreBeregning(behandling.id))
+      .then(() => next())
+      .finally(() => setIsLoading(false))
+  }
 
   const doedsdato = behandling.familieforhold.avdoede.opplysning.doedsdato
 
+  const visSøskenjustering =
```

Comment:
ø

Reply:
Skal det vises for både success og feil kun hvis det er 404?

Reply:
Vises ved suksess av åpenbare grunner
Vises ved feil=404 fordi det tidligere ikke er blitt gjort en søskenjustering.

Burde ikke returnere 404 Not Found for sistnevte tilfelle da, ettersom det vil gi logginnslag. Det er under arbeid, men tenkte denne PRen var en fin førsteiterasjon

Reply:
Aight